student/tests.py

```diff
@@ -105,3 +105,29 @@ def test_get_student_raises_exception_for_non_student_user():
     example_user_not_student.save()
     with pytest.raises(Student.DoesNotExist):
         Student.get_student(example_user_not_student.id)
+
+
+@pytest.mark.django_db
+def test_update_student_account_settings(example_student_a):
+    new_email = "guyyafe@gmail.com"
+    new_full_name = "guy y"
+    new_about = "I like tests"
+    example_student_a_data_edited = {"user": example_student_a.user, "full_name": new_full_name,
+                                     "email": new_email,
+                                     "date_of_birth": example_student_a.date_of_birth,
+                                     "phone_number": example_student_a.phone_number,
+                                     "educational_institution": example_student_a.educational_institution,
+                                     "major": example_student_a.major,
+                                     "about": new_about,
+                                     "graduation_date": example_student_a.graduation_date}
+    form = UpdateStudentAccountSettingsForm(instance=example_student_a, data=example_student_a_data_edited)
+    if form.is_valid():
+        form.save()
+    assert Student.objects.get(user_id=example_student_a.user.id).email == new_email
+    assert Student.objects.get(user_id=example_student_a.user.id).full_name == new_full_name
+    assert Student.objects.get(user_id=example_student_a.user.id).about == new_about
+    # Test inserting bad email, form should be invalid
+    bad_email = "guy"
+    example_student_a_data_edited["email"] = bad_email
+    form = UpdateStudentAccountSettingsForm(instance=example_student_a, data=example_student_a_data_edited)
+    assert form.is_valid() is False
```

Comment:
This is another test where you insert invalid data, please create another test for this

Reply:
fixed